test/firefox_user.js

```diff
@@ -1,5 +1,6 @@
-// Lift the default max 20 workers limit to something higher to avoid hangs when page needs to spawn a lot of threads.
-user_pref("dom.workers.maxPerDomain", 100);
+// Old Firefox browsers have a maxPerDomain limit of 20. Newer Firefox browsers default to 512. Match the new
+// default here to help test spawning a lot of threads also on older Firefox versions.
+user_pref("dom.workers.maxPerDomain", 512);
```

Comment:
Can we just delete this instead, to keep the config minimal?

Reply:
It wouldn't prevent testing would it, it would just mean that the max workers using in the test was whatever firefox intended in that particular version.

I think honoring the browser-specified `maxPerDomain` makes sense, since then we are testing what most users of firefix will have.

Also, for your CI your will be tests with recent firefox right?  So it won't slow you down to just delete this line will it?

Reply:
If the specific `test_pthread_hardware_concurrency` fails on older firefox then perhaps it should just `skip` in that case, since the test is testing something that the browser doesn't support (without a config mod anyway).

Reply:
> This sounds like maybe a real bug in older versions of FF then? What happens to users of older FF who run real wasm programs with thread pools? Is there something we can do to prevent the hanging in the wild?

Yes, it was. They would experience a hang too.

But we don't have a good way to know if `new Worker()` will never actually proceed. I had a spec bug about that somewhere to get a proper `new Worker()` or throw on ENORESOURCES, but it never went forward.

End users will need to manage their own pools so that they don't attempt to overspawn Workers. Fortunately end users rarely have to run posixtest suite for example to hammer a lot of thread creation.

Reply:
> i.e. it sounds like we need to somehow expose maxPerDomain to the user so that they avoid spawning more threads than that? But I guess its not visible to JS?

Unfortunately not. :/